hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java

```diff
@@ -108,20 +109,22 @@ private JetBootstrap() {
 
     static synchronized void executeJar(@Nonnull Supplier<JetInstance> supplier,
                            @Nonnull String jar, @Nullable String snapshotName,
-                           @Nullable String jobName, @Nonnull List<String> args
+                           @Nullable String jobName, @Nullable String mainClass, @Nonnull List<String> args
     ) throws Exception {
         JetBootstrap.jarName = jar;
         JetBootstrap.snapshotName = snapshotName;
         JetBootstrap.jobName = jobName;
         JetBootstrap.supplier = new ConcurrentMemoizingSupplier<>(() -> new InstanceProxy(supplier.get()));
 
         try (JarFile jarFile = new JarFile(jar)) {
-            if (jarFile.getManifest() == null) {
-                error("No manifest file in " + jar);
-            }
-            String mainClass = jarFile.getManifest().getMainAttributes().getValue("Main-Class");
-            if (mainClass == null) {
-                error("No Main-Class found in manifest");
+            if (StringUtil.isNullOrEmpty(mainClass)) {
+                if (jarFile.getManifest() == null) {
+                    error("No manifest file in " + jar);
```

Comment:
We can hint here that one can use `-c` option. Also on line 126.

Reply:
After I gave it a thought, if user is using the JetBootstrap directly, then it might not be meaningful to say `use -c option`. WDYT ?

Reply:
How do you mean using JetBootstrap directly?

Reply:
Never mind, I've missed the fact that the executeJar method is package private.